src/app/@ansyn/plugins/openlayers/open-layers-north-calculations/plugin/north-calculations-plugin.ts

```diff
@@ -71,32 +84,92 @@ export class NorthCalculationsPlugin implements IMapPlugin {
-			observables.push(iMap.projectionService.projectAccurately(point, iMap));
+			observables.push(this.ActiveMap.projectionService.projectAccurately(point, this.ActiveMap));
 		});
 
 		return Observable.forkJoin(observables);
 	}
 
-	setCorrectedNorth(iMap, retryNumber = 0): Promise<number> {
+	setCorrectedNorth(retryNumber = 0): Promise<number> {
 		if (retryNumber === this.maxNumberOfRetries) {
 			return;
 		}
 
-		return this.getCorrectedNorthOnce(iMap).toPromise().then((northData: INorthData) => {
-			iMap.mapObject.getView().setRotation(northData.actualNorth);
-			iMap.mapObject.renderSync();
+		return this.getCorrectedNorthOnce().toPromise().then<number>((northData: INorthData) => {
+			this.ActiveMap.mapObject.getView().setRotation(northData.actualNorth);
+			this.ActiveMap.mapObject.renderSync();
 			if (Math.abs(northData.northOffsetDeg) < this.thresholdDegrees) {
 				return northData.actualNorth;
 			}
-			return this.setCorrectedNorth(iMap, retryNumber + 1);
+			return this.setCorrectedNorth(retryNumber + 1);
 		}, (result) => {
 			return Promise.reject(result);
 		});
 	}
+
+	initPluginSubscribers() {
+		const pointNorth = this.actions$
+			.ofType<DisplayOverlaySuccessAction>(OverlaysActionTypes.DISPLAY_OVERLAY_SUCCESS)
+			.filter((action: DisplayOverlaySuccessAction) => action.payload.mapId === this.communicator.id)
+			.withLatestFrom(this.store$.select(statusBarStateSelector), ({ payload }: DisplayOverlaySuccessAction, { comboBoxesProperties }: IStatusBarState) => {
+				return [payload.ignoreRotation, this.communicator, comboBoxesProperties.orientation, payload.overlay];
+			})
+			.filter(([ignoreRotation, communicator]: [boolean, CommunicatorEntity, CaseOrientation, Overlay]) => Boolean(communicator) && communicator.activeMapName !== 'disabledOpenLayersMap')
+			.switchMap(([ignoreRotation, communicator, orientation, overlay]: [boolean, CommunicatorEntity, CaseOrientation, Overlay]) => {
+				return Observable.fromPromise(this.pointNorth(communicator))
+					.do(virtualNorth => {
+						communicator.setVirtualNorth(virtualNorth);
+						if (!ignoreRotation) {
+							switch (orientation) {
+								case 'Align North':
+									communicator.setRotation(virtualNorth);
+									break;
+								case 'Imagery Perspective':
+									communicator.setRotation(overlay.azimuth);
+									break;
+							}
+						}
+					});
+			}).subscribe();
+
+		const backToWorldSuccessSetNorth = this.actions$
+			.ofType<BackToWorldSuccess>(CoreActionTypes.BACK_TO_WORLD_SUCCESS)
+			.withLatestFrom(this.store$.select(statusBarStateSelector))
+			.do(([{ payload }, { comboBoxesProperties }]: [BackToWorldView, IStatusBarState]) => {
+				this.communicator.setVirtualNorth(0);
+				switch (comboBoxesProperties.orientation) {
+					case 'Align North':
+					case 'Imagery Perspective':
+						this.communicator.setRotation(0);
+				}
+			}).subscribe();
+
+		this.subscribers.push(pointNorth, backToWorldSuccessSetNorth)
```

Comment:
You should probably unsubscribe when disposing this plugin.

Reply:
please see `dispose()` function